### ISSUE TITLE
Inconsistent documentation for "heterogeneous environments"

### ISSUE THREAD

**Issue by MarkZ3** [icecc/icecream#398]
In the Readme, the section here: https://github.com/icecc/icecream#using-icecream-in-heterogeneous-environments
It mentions:
> This requires that the icecream daemon runs as root.

However, if I try to run iceccd (1.1.0 and master) when logged-in as root, it automatically switches to user icecc. If I try to explicitly specify the user root with the -u argument:
```
$ ./iceccd -u root
Error: -u <username> must not be root
```

BTW, I was trying this because for some reason icecc works with clang but not with gcc so I was trying to generate and specify the environment explicitly.

**Comment by HenryMiller1** [comment]
yeah, that section seems like it is obsolete.  I can't quite figure out when you would need to run as root anymore, if at all.

I recommend you follow the cross compiling sections, they are more complete.  You probably need to combine information from all sections to figure this out.  If you do figure it out, please update the documentation to be more helpful. Writing documentation tends to be hard for programmers and we could use the help.

One thing I know is you want to use icecc-create-environment to create your environment files.
